ipaserver/plugins/certmap.py

```diff
@@ -91,6 +91,53 @@
 register = Registry()
 
 
+def check_maprule_is_for_trusted_domain(api_inst, options, entry_attrs):
+    """
+    Check that the certmap rule references altSecurityIdentities and
+    associateddomain in options is a trusted domain.
+
+    :param api_inst: API instance
+    :param options: options passed to the command
+                    at least ipacertmapmaprule and
+                    associateddomain options are expected for the check
+
+    :raises: ValidationError if altSecurityIdentities is present in the
+             rule but no Active Directory trusted domain listed in
+             associated domains
+    """
+    is_trusted_domain_required = False
+    # If no explicit option passed, fallback to the content of the entry
+    # This helps to catch cases when an associated domain value is removed
+    # while the rule requires its presence
+    maprule = options.get('ipacertmapmaprule',
+                          entry_attrs.get('ipacertmapmaprule'))
+    if maprule:
+        if 'altsecurityidentities' in maprule.lower():
+            is_trusted_domain_required = True
+
+    if is_trusted_domain_required:
+        domains = options.get('associateddomain',
+                              entry_attrs.get('ipacertmapmaprule'))
```

Comment:
Should be entry_attrs.get('associateddomain')

Reply:
@flo-renaud thanks, silly copy/paste mistake. Fixed.